test/fipp/benchmark.clj

```diff
@@ -1,26 +1,42 @@
 (ns fipp.benchmark
-  (:require [clojure.pprint]
+  (:require [fipp.clojure]
             [fipp.edn]
-            [fipp.clojure]
-            [criterium.core :refer [bench]]))
+            [clojure.java.io :as io]
+            [clojure.pprint]
+            [clojure.test.check.generators :as gen]
+            [clojure.tools.reader.edn :as edn]
+            [criterium.core :refer [bench]])
+  (:import [java.io File Writer]))
 
-(def benched-fns [
-  ;prn
-  fipp.edn/pprint
-  fipp.clojure/pprint
-  ;clojure.pprint/pprint
-])
+(set! *warn-on-reflection* true)
 
-(def samples {
-  :long (vec (range 10000))
-  ;; Dirty hacky source of test data:
-  :mixed (read-string {:read-cond :allow} (str "[" (slurp "src/fipp/engine.cljc") "]"))
-})
+(def benched-fns [prn
+                  fipp.edn/pprint
+                  fipp.clojure/pprint
+                  clojure.pprint/pprint])
+
+(def writer
+  (proxy [Writer] []
+    (write [_])
+    (flush [])
+    (close [])))
+
+(defn samples []
+  (let [^File file (io/file ".benchmark-samples")]
+    (when-not (.exists file)
+      (spit file (with-out-str
+                   (fipp.edn/pprint (gen/sample gen/any-printable 1000)))))
```

Comment:
I didn't read this carefully enough, I see you're saving the samples file. Makes sense, but I still need to give more thought to whether we want to test random stuff or not.